### PR TITLE
Add PWD and better cd

### LINKED ISSUE #15: pwd

### PR DESCRIPTION
**Pull request by nikklassen**
closes #15

### CODE REVIEW

File: src/builtin.rs [hunk 1]
```diff
@@ -18,15 +21,54 @@ fn get_builtin(cmd : &str) -> Option<&Builtin> {
...
+    normalized_path
+}
+
 pub fn cd(args : &[&str]) -> io::Result<i32> {
-    let path = Path::new(args[0]);
-    env::set_current_dir(path).and(Ok(0))
+    fn change_to(p: &PathBuf) -> io::Result<()> {
+        let new_pwd_buf = normalize_logical_path(p);
+        env::set_var("PWD", &new_pwd_buf);
+        env::set_current_dir(new_pwd_buf)
```
Comment (danReynolds) [inline src/builtin.rs]: Our convention is to borrow by default, switch to passing a reference?

Comment (nikklassen) [inline src/builtin.rs]: Isn't that what I'm doing?

Comment (danReynolds) [inline src/builtin.rs]: You're passing `new_pwd_buf` to `set_current_dir`, won't it not come back from that?

Comment (nikklassen) [inline src/builtin.rs]: I should've paid more attention to which line you commented on.  I'll fix that.

**Pull request merged**